streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -303,7 +303,8 @@ public void computeTaskLags(final UUID uuid, final Map<TaskId, Long> allTaskEndO
     public long lagFor(final TaskId task) {
         final Long totalLag = taskLagTotals.get(task);
         if (totalLag == null) {
-            throw new IllegalStateException("Tried to lookup lag for unknown task " + task);
+            throw new IllegalStateException("Tried to lookup lag for unknown task: " + task
+                + " (This exception may be caused by that you don't call KafkaStreams#cleanUp when topology optimization is enabled)");
```

Comment:
Can we drop this comment? While it's definitely true, it's possible this might be caused by other reasons and I wouldn't want someone to jump the gun and reset their application unnecessarily.
If someone hits this, they'll probably open a file or send a question to the mailing list and then we can follow up with them to see if this might be the case

Reply:
you are right. I will revert it.